modules/internal/String.chpl

```diff
@@ -1068,29 +1068,28 @@ module String {
 
         var readIdx = 0;
         var writeIdx = 0;
+        var nbytes = c_int;
         while readIdx < localThis.buffLen {
-          var cp: int(32);
-          var nbytes: c_int;
           var multibytes = (localThis.buff + readIdx): c_string;
-          var maxbytes = (localThis.buffLen - readIdx): ssize_t;
-          const decodeRet = qio_decode_char_buf_esc(cp, nbytes, multibytes,
-                                                    maxbytes);
-          if (0xdc80<=cp && cp<=0xdcff) {
-            buff[writeIdx] = (cp-0xdc00):byteType;
-            writeIdx += 1;
-          }
-          else if (decodeRet != 0) {
-            // the string contains invalid data
+          var allowEsc = true;
+          
+           const (cp, nbytes) =  decodeHelp(localThis.buff, 
```

Comment:
This was the place I had a doubt with , Since we ignored decode error , so no special need to catch the error? and reducing two `bufferMemcpyLocal()` into one.

Reply:
I think I understand your doubt better now.

I think the comment below has some value and may help avoid some confusion. So I think we should still have a try/catch, where the catch block contains the same lines as the else block and that comment. That would be code repetition, but I think it can help whoever may read this code. Does that make sense?

Reply:
Also, wrong indentation in this line